### ISSUE TITLE
False positive for "Enum" has no attribute "__iter__"; maybe "__str__" or "__dir__"? (not iterable)

### ISSUE THREAD

**Issue by ImreC** [python/mypy#11099]
**Bug Report**

Mypy seems to give a false positive when trying to iterate over an Enum passed to a method.

**To Reproduce**

```python
def return_names_for_enum(some_enum: Enum):
    return [x.name for x in some_enum]
```

**Expected Behavior**
Since this all works: 
```python
from enum import Enum

class Test(Enum):
    test = 'OK'

names = [x.name for x in Test]
```

I also expect this to work when done in a method.

**Actual Behavior**

`"Enum" has no attribute "__iter__"; maybe "__str__" or "__dir__"? (not iterable)`

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: "--ignore-missing-imports"
- Python version used: 3.8.10
- Operating system and version: Ubuntu 20.04 LTS (Pop!_Os 20.04)

**Comment by erictraut** [comment]
You need to annotate the type of parameter `some_enum` to `Type[Enum]`. If you correct that annotation, this type checks without error.

**Comment by ImreC** [comment]
Thanks for the info. I'll have a look at the docs and close this.
